### PR TITLE
feat: allow to pass arguments to docker

### PR DESCRIPTION
**Pull request by OmriBarZik**
# Summary
Added the ability to pass dockly's arguments to [lirantal/dockly](https://hub.docker.com/r/lirantal/dockly) docker image without calling dockly explicitly

before
```bash
docker run -it --rm -v /var/run/docker.sock:/other/place/docker.sock lirantal/dockly node index -s /other/place/docker.sock 
```

now
```bash
docker run -it --rm -v /var/run/docker.sock:/other/place/docker.sock lirantal/dockly -s /other/place/docker.sock 
```

## Proposed Changes

  - Added the ability to pass dockly docker arguments without calling dockly explicitly
  
## Checklist

- [ ] I added tests
- [ ] I updated the README if necessary
- [ ] This PR introduces a breaking change
- [ ] Fixed issue #
- [ ] I added a picture of a cute animal cause it's fun

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,4 +13,7 @@ WORKDIR /app
 ENV NODE_ENV production
 RUN yarn install --frozen-lockfile
 
+COPY docker-entrypoint.sh /usr/local/bin/
```
Comment (lirantal) [inline Dockerfile:16]: OH, It looks like we were missing docker-entrypoint.sh in the built image?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
+if [ "${1#-}" != "${1}" ] || [ -z "$(command -v "${1}")" ]; then
```
Comment (lirantal) [inline docker-entrypoint.sh:4]: That bash magic 😁

File: Dockerfile [hunk 3]
```diff
@@ -13,4 +13,7 @@ WORKDIR /app
...
+ENTRYPOINT ["docker-entrypoint.sh"]
```
Comment (lirantal) [inline Dockerfile:17]: Why not just `ENTRYPOINT ["node", "index.js"]` and then we pass in the `-s ...` extra arguments?

**Review by lirantal — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (OmriBarZik) [reply Dockerfile:17]: what if you want to control the docker container itself and not dockly specifically, the `docker-entrypoint.sh` script enables that

File: docker-entrypoint.sh (hunk 2 above)
Comment (OmriBarZik) [reply docker-entrypoint.sh:4]: There are 2 checks in this bash script.
the first check checks if there are `-` in the first argument like so:
```bash
1=foo
echo $1     # output: foo
echo ${1#-} # output: foo

1=--foo       
echo $1     # output: --foo
echo ${1#-} # output: -foo
```
the second check checks if the first argument is a command that exists in the system.
the flag `-z` check if there is an output. if the command is found on the system `command -v "${1}"` will return its path (!true) if not it will return nothing (!false).

TLDR:
`IF` (is the first character in the first argument is `-`) `OR` (`NOT`(is the first argument a command)) `THEN` run dockly

File: Dockerfile (hunk 3 above)
Comment (lirantal) [reply Dockerfile:17]: when you say control you mean to spawn a shell for debugging purposes? if so, yeah you'd need to rebuild the image without the entrypoint, which I think is ok to do as it's not something we'd need to do constantly and easier to maintain. WDYT?

File: docker-entrypoint.sh (hunk 2 above)
Comment (lirantal) [reply docker-entrypoint.sh:4]: Splendid, thanks!

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
```
Comment (lirantal) [inline docker-entrypoint.sh:3]:
```suggestion
# There are 2 checks in this bash script.
# the first check checks if the first character in the first argument is - like so:
#
# 1=foo
# echo $1     # output: foo
# echo ${1#-} # output: foo
#
# 1=--foo       
# echo $1     # output: --foo
# echo ${1#-} # output: -foo
# the second check checks if the first argument is a command that exists in the system.
# the flag -z check if there is an output. if the command is found on the system command -v "${1}" will return its # # path (!true) if not it will return nothing (!false).

# TLDR:
# IF (is the first character in the first argument is -) OR (NOT(is the first argument a command)) THEN run dockly
# 
# WHY: we do this to enable users to run commands on the container such as opening a shell
```

Comment (lirantal) [reply docker-entrypoint.sh:3]: Added comments to this script for our future selves

File: Dockerfile (hunk 3 above)
Comment (OmriBarZik) [reply Dockerfile:17]: Ok, I agree. I'll remove it

**Comment by OmriBarZik** [comment]
What do we want to do with this PR?

**Comment by lirantal** [comment]
Sorry for the delay Omri, it looks good with the change as-is now. I didn't notice you updated the files.

Good to merge on my end! 👍🏼

**Review by lirantal — APPROVED** [review]

**Comment by OmriBarZik** [comment]
GitHub is telling me that I cant merge this PR, @lirantal do you know what we can do?

**Comment by lirantal** [comment]
Checking why...

**Comment by lirantal** [comment]
Looks like commits in this pull request aren't verified. I'll merge it for now but maybe you can set it up later on in your free time for future PRs? (also, it's a good practice regardless :-))

image
